rir/src/compiler/translations/rir_2_pir/stack_machine.cpp

```diff
@@ -142,23 +143,33 @@ void StackMachine::runCurrentBC(Rir2Pir& rir2pir, Builder& insert) {
             }
             rir::Code* promiseCode = srcFunction->codeAt(argi);
             Promise* prom = insert.function->createProm();
+            RirCompiler& cmp = rir2pir.compiler();
             {
+                Builder* oldBuilder = cmp.getBuilder();
                 Builder promiseBuilder(insert.function, prom);
-                Rir2Pir compiler(rir2pir.compiler(), promiseBuilder,
-                                 srcFunction, promiseCode);
-                compiler.translate();
+                cmp.setBuilder(&promiseBuilder);
```

Comment:
sorry, but I really do not agree with this! I think swapping the builder back and forth like this is really brittle. I just see a lot more ways to get it wrong and no benefit. Think of it in terms of invariants. Can you formulate the  invariant of the compiler object? It's something like: "While a Rir2Pir object that points to a RirCompiler object is executing it's translate function, the builder field of the compiler object points to a builder that corresponds to the the code being compiled by the translate function of the rir2pir object, but temporarily it can also point to another builder, but will be set back, before the outer translate function regains control."